src/views/Farms/components/StakedLP.tsx

```diff
@@ -23,13 +30,29 @@ const StakedLP: React.FunctionComponent<React.PropsWithChildren<StakedLPProps>>
   tokenAmountTotal,
   quoteTokenAmountTotal,
 }) => {
+  const dispatch = useAppDispatch()
+  const { chainId } = useActiveWeb3React()
+  const pendingFarm = useNonBscFarmPendingTransaction(lpAddress)
+  const [onPresentTransactionModal] = useModal(<WalletModal initialView={WalletView.TRANSACTIONS} />)
+
   const displayBalance = useMemo(() => {
     return formatLpBalance(stakedBalance)
   }, [stakedBalance])
 
+  const onClickLoadingIcon = () => {
+    if (pendingFarm.length > 0) {
+      onPresentTransactionModal()
+    } else {
+      dispatch(pickFarmTransactionTx({ tx: pendingFarm[0].txid, chainId }))
```

Comment:
if `pendingFarm.length` is `== 0`, it would throw error here, right?

Reply:
Updated